c2rust-analyze/src/main.rs

```diff
@@ -641,9 +713,24 @@ fn run(tcx: TyCtxt) {
 
     let mut gasn =
         GlobalAssignment::new(gacx.num_pointers(), PermissionSet::UNIQUE, FlagSet::empty());
+    for (ptr, info) in gacx.ptr_info().iter() {
+        if info.contains(PointerInfo::REF) {
+            gasn.flags[ptr].insert(FlagSet::FIXED);
+        }
+    }
+
     for info in func_info.values_mut() {
         let num_pointers = info.acx_data.num_pointers();
-        let lasn = LocalAssignment::new(num_pointers, PermissionSet::UNIQUE, FlagSet::empty());
+        let mut lasn = LocalAssignment::new(num_pointers, PermissionSet::UNIQUE, FlagSet::empty());
+
+        for (ptr, info) in info.acx_data.local_ptr_info().iter() {
+            if info.contains(PointerInfo::REF)
+                && !info.contains(PointerInfo::RECOGNIZED_TEMPORARY_REF)
```

Comment:
Could you put this in a `fn PointerInfo::is_fixed` method?  I think it would be clearer right next to the `PointerInfo` definition.

Reply:
I think that would be confusing, since just above this is the global case that sets `FIXED` for any `REF`, regardless of whether it was `RECOGNIZED` or not.

Reply:
Oh, true, if the conditions are different for global ptrs.  But why is `FIXED` being set for all `REF`s globally?  Why don't we skip temporaries there, too?

Reply:
Actually, after looking at this more, I think there are some adjustments I want to make to the rules for applying `FIXED`.  This will involve unifying the local and global cases, at which point it would make sense to apply a change like you suggested.